ocaml/xapi/xapi_host.ml

```diff
@@ -1403,7 +1403,12 @@ let install_server_certificate ~__context ~host ~certificate ~private_key
   (* sever all connections done with stunnel *)
   Xapi_mgmt_iface.reconfigure_stunnel ~__context
 
-let emergency_reset_server_certificate ~__context =
+let reset_server_certificate ~__context ~host =
+  let self = Helpers.get_localhost ~__context in
+  ( if host <> self then
+      let msg = "reset_server_certificate host != localhost" in
+      raise Api_errors.(Server_error (internal_error, [msg]))
+  ) ;
```

Comment:
I don't think this check belongs to this layer, it should be contained to the `message_forwarding` module.

Reply:
This assertion does no harm, but we can rely on the forwarding layer to have done the right thing.